src/Controls/src/Core/ImageButton/ImageButton.iOS.cs

```diff
@@ -0,0 +1,54 @@
+#nullable disable
+using System;
+using CoreGraphics;
+using Microsoft.Maui.Graphics;
+using UIKit;
+
+namespace Microsoft.Maui.Controls
+{
+	public partial class ImageButton : ICrossPlatformLayout
+	{
+		/// <summary>
+		/// Measure the desired size of the ImageButton based on the image size taking into account the padding.
+		/// </summary>
+		/// <param name="widthConstraint"></param>
+		/// <param name="heightConstraint"></param>
+		/// <returns>Returns a <see cref="Size"/> representing the width and height of the ImageButton.</returns>
+		/// <remarks>This method is used to override the SizeThatFitsImage() on wrapper view.</remarks>
+		Size ICrossPlatformLayout.CrossPlatformMeasure(double widthConstraint, double heightConstraint)
+		{
+			if (Handler?.PlatformView is not UIButton platformButton)
+				return Size.Zero;
+
+			Thickness padding = Padding.IsNaN ? default(Thickness) : Padding;
+
+			Thickness contentEdgeInsets = new Thickness
+			{
+				Left = padding.Left,
+				Right = padding.Right,
+				Top = padding.Top,
+				Bottom = padding.Bottom
+			};
+
+			CGSize imageSize = platformButton.CurrentImage?.Size ?? CGSize.Empty;
+
+			double contentWidth = imageSize.Width + contentEdgeInsets.Left + contentEdgeInsets.Right;
+			double contentHeight = imageSize.Height + contentEdgeInsets.Top + contentEdgeInsets.Bottom;
```

Comment:
[Thickness](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.thickness?view=net-maui-9.0) has a [HorizontalThickness](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.thickness.horizontalthickness?view=net-maui-9.0) and [VerticalThickness](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.thickness.verticalthickness?view=net-maui-9.0) properties you could use.